### DISCUSSION TITLE
Nginxのサイマル配信を検証する

### DISCUSSION THREAD

**Discussion by TakumaNakagame** [cloudnativedaysjp/dreamkast-broadcast#170] [Ideas]
## ゴール
Nginxを使ったサイマル配信が適切に行える

- OBSに適切に映像が送られる
- OBSのシーン切り替え時にブラックアウトなどなく、スムーズに切り替えできる
- `nginx:obs = 1:3` 環境で実用できる
- できるだけ低遅延
- Nginxの負荷が高くないか

### 注意点
- シーン切り替え時に、「ブラックアウト」や「数秒前の映像が一瞬出てくる」ことがある

### OP/EDのサイマル配信
- 送信元： 現地スタジオのOBS
- 送信先： Switcher全台

現地の映像をすべてのトラックで再生する

### リモート登壇
- 送信元： StreamYard
- 送信先： switcher個別

StreamYardでの登壇映像をNginxを介してOBSに届ける。
かつ、映像は特定トラックのみで配信される。

### バージョン情報
- OBS: `27.2.4` https://github.com/obsproject/obs-studio/releases/tag/27.2.4

### Nginxに映像を送る方法
OBSの配信設定より以下を設定する

- サービス： `カスタム`
- サーバー： `rtmp://nginx0{1,2}.cloudnativedays.jp:10002/live`
- ストリームキー：任意の重複しない値

### 配信された映像をVLCで見る
VLCメディアプレーヤーの「メディア」→「ネットワークストリームを開く」を選択

- ネットワークURL： `rtmp://nginx0{1,2}.cloudnativedays.jp:10002/live/{ストリームキー}`  
  例： `rtmp://nginx01.cloudnativedays.jp:10002/live/kameneko`

### 配信された映像をOBSで見る
ソースより、「メディアソース」を選択

- ローカルファイル： `無効`
- ソースがアクティブになったときに再生を再開する： `無効`
- ネットワークバッファリング： `2MB`
- 入力： `rtmp://nginx0{1,2}.cloudnativedays.jp:10002/live/{ストリームキー}`  
  例： `rtmp://nginx01.cloudnativedays.jp:10002/live/kameneko`
- 入力フォーマット： `rtmp`
- 再接続の遅延： `10 S`
- 可能な場合はハードウェアデコードを使用： `無効`
- 再生終了時に何も表示しない： `有効`
- アクティブでないときにファイルを閉じる： `無効`
- YUV色範囲： `自動`
- 線形空間にアルファを適用する： `無効`
- シーク可能： `無効`

### 接続先
各パスワードはいつものやつ
- OBS
  - `switcher{1,2,3}.cloudnativedays.jp`
- nginx
  - 表LAN： `nginx0{1,2,3}.cloudnativedays.jp`
  - 裏LAN： `192.168.71.2{1,2,3}`

**Comment by capsmalt** [comment]
構成:
手元のOBS -> nginx -> switcher0{1,2,3}

- [x] OBSに適切に映像が送られる

- [x] OBSのシーン切り替え時にブラックアウトなどなく、スムーズに切り替えできる

- [x] nginx:obs = 1:3 環境で実用できる

- [ ] できるだけ低遅延

- [ ] Nginxの負荷が高くないか

**Reply by capsmalt** [reply]
・シーン切り替えのブラックアウトを確認できなかった。イメージしてるものが異なるかもしれない

・mac -- vnc --> switcher 接続時、switcher内のUI上の動きが鈍くて操作しづらい
